taskcat/_cli_modules/test.py

```diff
@@ -103,9 +230,10 @@ def retry(
         )
 
     # pylint: disable=too-many-locals,too-many-branches,too-many-statements
-    @staticmethod  # noqa: C901
+    @classmethod  # noqa: C901
     # pylint: disable=too-many-arguments
     def run(  # noqa: C901
+        cls,
```

Comment:
This did not work as I hoped. 

It's hard to see in the git diff but if you look at the file its cleaned up the logic and made it pretty straight forward.